install/minikube-installation-guide.md

```diff
@@ -0,0 +1,236 @@
+
+This installation guide may not work for `MacOS` installations of `Minikube`, due to the lack of
+nested virtualization support on that platform.
+
+This installation guide has not been tested on a `Windows` installation.
+
+See the [`Minikube` installation guide](https://kubernetes.io/docs/tasks/tools/install-minikube/)
+for details on installation, whilst paying attention to the following [Setup](#setting-up-minikube]
+section of this document. Also pay attention to the following
+[Nested Virtualization](#checking-for-nested-virtualization) section of this document, as
+`Kata Containers` will not function under `Minikube` if this is not enabled on your system.
+
+## Setting up `Minikube`
+
+To enable `Kata Containers` under `Minikube`, you need to add a few configuration options to the
+default `Minikube` setup. This is easily accomplished, as `Minikube` supports them on the setup commandline.
+
+Here are the features, and why you need them:
+
+| what | why |
+| ---- | --- |
+| --vm-driver kvm2 | The host VM driver |
+| --memory 6144 | Allocate sufficient memory, as Kata containers default to 1 or 2Gb |
+| --network-plugin=cni | As recommended for [minikube CRI-o](https://kubernetes.io/docs/setup/minikube/#cri-o) |
+| --enable-default-cni | As recommended for [minikube CRI-o](https://kubernetes.io/docs/setup/minikube/#cri-o) |
+| --container-runtime=cri-o | Using CRI-O for Kata |
+| --bootstrapper=kubeadm | As recommended for [minikube CRI-o](https://kubernetes.io/docs/setup/minikube/#cri-o) |
+
+> **Note:** This example deploys `Kata Containers` using `CRI-O`. The same procedure should also work
+> for a `containerd` based `k8s` installation, but is not covered in this document.
+
+> **Note:** Prior to `Minikube`/`Kubernetes` v1.14, the beta `RuntimeClass` feature also needed enabling with
+> the following. Recent `Minikube`/`Kubernetes` installations should no longer require this setting.
+>
+> | what | why |
+> | ---- | --- |
+> | --feature-gates=RuntimeClass=true | Kata needs to use the RuntimeClass k8s feature |
+
+The full command is therefore:
+
+```sh
+$ minikube start --vm-driver kvm2 --memory 6144 --network-plugin=cni --enable-default-cni --container-runtime=cri-o --bootstrapper=kubeadm 
+```
+
+## Checking for nested virtualization
+
+For `Kata Containers` to work under a `Minikube` VM, your host system must support
+nested virtualization. If you are using a Linux system utilising Intel `VT-x` 
+and the `kvm_intel` driver, you can perform the check by:
+
+```sh
+$ cat /sys/module/kvm_intel/parameters/nested
+Y
+```
+
+If your system does not report `Y` from the `nested` parameter, then details on how
+to enable nested virtualisation can be found on the
+[KVM Nested Guests page](https://www.linux-kvm.org/page/Nested_Guests)
+
+## Check `Minikube` is running
+
+Before you install `Kata Containers`, perform a quick check that your `Minikube` is
+operating. On your guest:
+
+
+```sh
+$ kubectl get nodes
```

Comment:
Could this become something like?:

```bash
$ kubectl get nodes | grep -E "\<minikube\>" | grep -E "\<Ready\>"
```

Reply:
Oh, so `kubectl` will block on any command until it can talk to the cluster? OOI I wonder if there is either a formal `is-the-cluster-up-yet` command or a timeout / async option? If the cluster never comes up, does `get nodes` error / timeout or just hang forever?

Reply:
Yuck. Oh well...